<launch>

   <!-- Arguments that are passed from the top launchfile -->
   <!-- Vehicle-related arguments -->
   <arg name="vehicle_id" />
   <arg name="vehicle_ns" />

   <!-- Namespace-related arguments -->
   <arg name="objects_ground_truth_topic_with_ns" />
   <arg name="internal_communication_subns" />
   <arg name="global_communication_ns" />

   <!-- Topic-related arguments -->
   <arg name="car2x_topic1" default="dummy_car2x_topic1"/>
   <arg name="car2x_topic2" default="dummy_car2x_topic2"/>
   <arg name="car2x_topic3" default="dummy_car2x_topic3"/>
   <arg name="clear_queue_topic" default="default_communication_clear_queue_topic"/>

    <!-- LAUNCH NODES -->
    <!-- For Car2X-Topic 1 -->
    <node pkg="sim_sample_communication_ros_tool" type="comm_module" name="com_mgmt_$(arg car2x_topic1)" output="screen">

            <param name="communication_external_topic" value="$(arg global_communication_ns)/$(arg car2x_topic1)" />
            <param name="communication_internal_in_topic" value="/$(arg vehicle_ns)/$(arg internal_communication_subns)/in/$(arg car2x_topic1)" />
            <param name="communication_internal_out_topic" value="/$(arg vehicle_ns)/$(arg internal_communication_subns)/out/$(arg car2x_topic1)" />
            <param name="clear_queue_topic" value="$(arg clear_queue_topic)" />

            <!-- Vehicle-related -->
            <param name="vehicle_id" value="$(arg vehicle_id)" />
            <param name="vehicle_ns" value="$(arg vehicle_ns)" />

    </node>

    <!-- For Car2X-Topic 2 -->
    <node pkg="sim_sample_communication_ros_tool" type="comm_module" name="com_mgmt_$(arg car2x_topic2)" output="screen">

            <param name="communication_external_topic" value="$(arg global_communication_ns)/$(arg car2x_topic2)" />
            <param name="communication_internal_in_topic" value="/$(arg vehicle_ns)/$(arg internal_communication_subns)/in/$(arg car2x_topic2)" />
            <param name="communication_internal_out_topic" value="/$(arg vehicle_ns)/$(arg internal_communication_subns)/out/$(arg car2x_topic2)" />
            <param name="clear_queue_topic" value="$(arg clear_queue_topic)" />

            <!-- Vehicle-related -->
            <param name="vehicle_id" value="$(arg vehicle_id)" />
            <param name="vehicle_ns" value="$(arg vehicle_ns)" />

    </node>

    <!-- For Car2X-Topic 3 -->
    <node pkg="sim_sample_communication_ros_tool" type="comm_module" name="com_mgmt_$(arg car2x_topic3)" output="screen">

            <param name="communication_external_topic" value="$(arg global_communication_ns)/$(arg car2x_topic3)" />
            <param name="communication_internal_in_topic" value="/$(arg vehicle_ns)/$(arg internal_communication_subns)/in/$(arg car2x_topic3)" />
            <param name="communication_internal_out_topic" value="/$(arg vehicle_ns)/$(arg internal_communication_subns)/out/$(arg car2x_topic3)" />
            <param name="clear_queue_topic" value="$(arg clear_queue_topic)" />

            <!-- Vehicle-related -->
            <param name="vehicle_id" value="$(arg vehicle_id)" />
            <param name="vehicle_ns" value="$(arg vehicle_ns)" />

    </node>

</launch>
